New .Net Core client (#216)
infinispan/infinispan.github.io: _posts/2021-06-01-dotnet-core-client.adoc

---
layout: blog
title: An Infinispan .Net Core Client over Hotrod protocol
date:'2021-06-01T12:00:00.000-00:00'
author: rigazilla
tags: [ "hotrod", ".Net Core", "client" "C#" ]
---
Dear Infinispanners,

we'd like to share with you a new project the Infinispan Team is working on: https://github.com/infinispan/Infinispan.Hotrod.Core[Infinispan.Hotrod.Core].

> As you may know the current .Net client is based on the {cpp} core implementation, this architectural solution has pros and cons:
> features can be easily ported between {cpp} and C# (this has really guarateed features parity beetween the two clients over the time) but as a drawback:
> it ties the client to a specific architecture, complicating portability and distribution.

Reviewer: ```suggestion
Our current .NET client is based on the {cpp} core implementation, which is an solution that has its pros and cons.
It makes it easier to guarantee parity between the {cpp} and C# clients over the time, but has the drawback of tying clients to specific a architecture. 
In turn that complicates portability and distribution, making the release lifecycle for those clients more onerous and sluggish.
```

> The https://github.com/infinispan/Infinispan.Hotrod.Core[Infinispan.Hotrod.Core] is a 100% C# Hotrod client designed with the aim to be portable across

Reviewer: ```suggestion
The https://github.com/infinispan/Infinispan.Hotrod.Core[Infinispan.Hotrod.Core] is a 100% C# Hot Rod client designed with the aim of being portable across
```

> systems via the .Net Core platform and easy to deploy and consume via the Nuget platform.

Reviewer: ```suggestion
systems via the .Net Core platform and easier to deploy and consume via the Nuget platform.
```

> If you live in the .Net Core ecosystem you'll probably want to look at it. Entry points are:

Reviewer: ```suggestion
If you are involved with the .NET Core ecosystem we hope you will find this project of interest. Entry points for the project are:
```

* https://github.com/infinispan/Infinispan.Hotrod.Core[Infinispan.Hotrod.Core], current status and news;
* https://github.com/infinispan/Infinispan.Hotrod.Core/tree/main/Infinispan.Hotrod.Core.XUnitTest[XUnit test], this will contain the same testsuite of the

> current https://github.com/infinispan/dotnet-client[.Net client] project, so to ease maturity comparison;

Reviewer: ```suggestion
current https://github.com/infinispan/dotnet-client[.NET client] project with the aim of making it easier to compare with the .NET core client as it matures.
```

> * https://github.com/infinispan/Infinispan.Hotrod.Core/tree/main/Infinispan.Hotrod.Application[Sample Application], an example of project which uses the client package;

Reviewer: ```suggestion
* https://github.com/infinispan/Infinispan.Hotrod.Core/tree/main/Infinispan.Hotrod.Application[Sample Application], an example of project which uses the client package.
```

> * https://www.nuget.org/packages/Infinispan.Hotrod.Core/[Nuget], where the package is published.

Reviewer: ```suggestion
* https://www.nuget.org/packages/Infinispan.Hotrod.Core/[Nuget] is where the package is published.
```

Hope this will makes C# developers happy!

> Please let us know your thoughts, good place for them is the https://github.com/infinispan/Infinispan.Hotrod.Core/issues[issues] page.

Reviewer: ```suggestion
Please let us know your thoughts, a good place for them is the https://github.com/infinispan/Infinispan.Hotrod.Core/issues[issues] page.
```
